model/host/host_intent.go

```diff
@@ -113,13 +113,13 @@ func generateParentCreateOptions(pool *evergreen.ContainerPool) CreateOptions {
 }
 
 func MakeContainersAndParents(d distro.Distro, pool *evergreen.ContainerPool, newContainersNeeded int, hostOptions CreateOptions) ([]Host, []Host, error) {
-	// get the parents that are up and split into ones that already have a contaienr from this distro
+	// get the parents that are up and split into ones that already have a container from this distro
 	currentHosts, err := GetContainersOnParents(d)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "Could not find number of containers on parents")
 	}
 	maxImages := defaultMaxImagesPerParent
-	if pool != nil && pool.MaxImages > 0 {
+	if pool.MaxImages > 0 {
```

Comment:
Instead of removing the nil check can we move it to the top?

Reply:
I think it's okay for functions to assume that they've been passed non-nil input. We check this in each function that calls it so that will cover it.